cubic-server/generation/features/underground/OreVein.cpp

```diff
@@ -0,0 +1,253 @@
+                    } else {
+                        spawnRate = ((spawnTries * y) / midVal) - spawnTries;
+                    }
+                    if (block == Blocks::Stone::toProtocol() && !isSkippedWhenAirExposed({x, (y - abs(minY)), z}, spawnSize, skipRate) &&
+                        (utility::PseudoRandomGenerator::getInstance()->generateNumber(0.0, spawnTries) < spawnRate)) {
+                        positions.emplace_back(x, y - abs(minY), z);
+                        spawnTry--;
+                        x < z ? x = x + MAX_NB_OF_BLOCKS[spawnSize] : z = z + MAX_NB_OF_BLOCKS[spawnSize];
+                    }
+                }
+            }
+        }
+    }
+    return positions;
+}
+
+std::deque<Position>
+OreVein::defineAllBlobPositions(const GenerationType generationType, const int spawnSize, const int minY, const int maxY, const double skipRate, const double spawnTries)
+{
+    using namespace world_storage;
+    std::deque<Position> positions;
+
+    if (generationType == GenerationType::UNIFORM) {
+        positions = addPositions(spawnTries, minY, maxY, spawnSize, skipRate, UNIFORM_SPAWN_RATE);
+    } else if (generationType == GenerationType::TRIANGLE) {
+        positions = computeTriangleDistribution(spawnSize, spawnTries, minY, maxY, skipRate);
+    }
+    return positions;
+}
+
+void OreVein::createBlob(const BlockId &blockID, const int spawnSize, const Position &pos) const
+{
+    auto nbOfBlocksInBlob = utility::PseudoRandomGenerator::getInstance()->generateNumber(0, MAX_NB_OF_BLOCKS[spawnSize]);
+    int m = pos.x + spawnSize; /**< Longitudinal spread of the blob */
+    int n = pos.z + spawnSize; /**< Latitudinal spread of the blob */
+    int p = pos.y + spawnSize; /**< Altitudinal spread of the blob */
+    int r = spawnSize / 2; /**< Radius of the blob */
+    int customSkipRate;
+    int nb = 0;
+
+    for (int y = pos.y; y < p && nb < nbOfBlocksInBlob; y++) {
+        for (int x = pos.x; x < m; x++) {
+            for (int z = pos.z; z < n; z++) {
+                customSkipRate =
+                    abs(((((m / 2) - x) % world_storage::SECTION_WIDTH) ^ 2) + ((((p / 2) - y) % world_storage::SECTION_WIDTH) ^ 2) +
+                        ((((n / 2) - z) % world_storage::SECTION_WIDTH) ^ 2) % world_storage::SECTION_WIDTH);
+                auto block = _chunk.getBlock({x, y, z});
+                if ((x - pos.x < r && z - pos.z < r) && r / 2 > customSkipRate && nb < nbOfBlocksInBlob &&
+                    (block != Blocks::Air::toProtocol() && block != Blocks::Bedrock::toProtocol()) && y > world_storage::CHUNK_HEIGHT_MIN) {
+                    _chunk.updateBlock({x, y, z}, blockID);
+                    nb++;
+                }
+            }
+        }
+    }
+}
+
+void OreVein::generateIronBlobs()
+{
+    std::deque<Position> posTriangle = defineAllBlobPositions(GenerationType::TRIANGLE, BLOB_SPAWN_SIZE_IRON_TRIANGLE, -24, 56, 0, 10);
```

Comment:
please use constexpr int instead of magic values

Reply:
Ugh... Well, it not like I care if the .hpp is +100 lines full of variables :)

Reply:
Just put the constepxr value at the top of that cpp if you don't use them in other files.